core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -348,6 +350,13 @@ public void syncPostBatch(
             int shortCircuited = 0;
             final Map<BatchItem, CompletableFuture<Exception>> sendFutures = new HashMap<>();
             for (final BatchItem item : batch) {
+                if (item.getBrokerId() == null) {
+                    item.updateStatusAndDetail(EventPublishingStatus.FAILED,
+                            String.format("No leader for partition: %s, topic: %s.", item.getPartition(), topicId));
+                    LOG.warn("Failed to publish to kafka. No leader for partition [{}], topic [{topicId}].",
```

Comment:
in Kafka it is pair (topic, partition) it will be more readable in the logs. also for consistency I would use either `{}` or both named `{topicId}` and `{partitionId}`